internal/backend/runtime/omni/controllers/omni/talos_upgrade_status.go

```diff
@@ -448,13 +453,16 @@ func updateMachine(ctx context.Context, r controller.ReaderWriter, logger *zap.L
 	)
 }
 
-func createInitialTalosVersion(ctx context.Context, r controller.ReaderWriter, machine *omni.ClusterMachine, talosVersion, schematicID string) error {
+func createInitialTalosVersion(ctx context.Context, r controller.ReaderWriter, machine *omni.ClusterMachine, schematicConfig *omni.SchematicConfiguration) error {
 	res := omni.NewClusterMachineTalosVersion(resources.DefaultNamespace, machine.Metadata().ID())
 
 	helpers.CopyAllLabels(machine, res)
 
-	res.TypedSpec().Value.TalosVersion = talosVersion
-	res.TypedSpec().Value.SchematicId = schematicID
+	res.TypedSpec().Value.TalosVersion = schematicConfig.TypedSpec().Value.TalosVersion
+	res.TypedSpec().Value.SchematicId = schematicConfig.TypedSpec().Value.SchematicId
+	res.TypedSpec().Value.SchematicInitialized = schematicConfig.TypedSpec().Value.SchematicInitialized
+	res.TypedSpec().Value.SchematicInvalid = schematicConfig.TypedSpec().Value.SchematicInvalid
+	res.TypedSpec().Value.SecureBootStatus = schematicConfig.TypedSpec().Value.SecureBootStatus
```

Comment:
Not sure about this forwarding.
`TalosUpgradeStatusController` still has `MachineStatus` as the input, so maybe we can read it from there directly instead of adding the same fields to 2 types of resources?

Reply:
ah cool, that makes sense, also makes things easier. will do.

Reply:
done